bindings/src/main/java/io/pravega/storage/filesystem/FileSystemChunkStorage.java

```diff
@@ -0,0 +1,294 @@
+        return Files.size(path);
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        try {
+            PosixFileAttributes attrs = Files.readAttributes(Paths.get(config.getRoot(), chunkName),
+                    PosixFileAttributes.class);
+            ChunkInfo information = ChunkInfo.builder()
+                    .name(chunkName)
+                    .length(attrs.size())
+                    .build();
+
+            return information;
+        } catch (IOException e) {
+            throw  convertExeption(chunkName, "doGetInfo", e);
+        }
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        try {
+            FileAttribute<Set<PosixFilePermission>> fileAttributes = PosixFilePermissions.asFileAttribute(FileSystemUtils.READ_WRITE_PERMISSION);
+
+            Path path = Paths.get(config.getRoot(), chunkName);
+            Path parent = path.getParent();
+            assert parent != null;
+            Files.createDirectories(parent);
+            Files.createFile(path, fileAttributes);
+
+        } catch (IOException e) {
+            throw convertExeption(chunkName, "doCreate", e);
+        }
+
+        return ChunkHandle.writeHandle(chunkName);
+    }
+
+    private ChunkStorageException convertExeption(String chunkName, String message, Exception e) {
+        if (e instanceof FileNotFoundException || e instanceof NoSuchFileException) {
+            return new ChunkNotFoundException(chunkName, message, e);
+        }
+        if (e instanceof FileAlreadyExistsException) {
+            return  new ChunkAlreadyExistsException(chunkName, message, e);
+        }
+        return new ChunkStorageException(chunkName, message, e);
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) throws IllegalArgumentException {
+        return Files.exists(Paths.get(config.getRoot(), chunkName));
+    }
+
+    @Override
+    protected void doDelete(ChunkHandle handle) throws ChunkStorageException, IllegalArgumentException {
+        try {
+            Files.delete(Paths.get(config.getRoot(), handle.getChunkName()));
+        } catch (IOException e) {
+            throw convertExeption(handle.getChunkName(), "doDelete", e);
+        }
+    }
+
+    @Override
+    protected ChunkHandle doOpenRead(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        Path path = Paths.get(config.getRoot(), chunkName);
+
+        if (!Files.exists(path)) {
+            throw new ChunkNotFoundException(chunkName, "FileSystemChunkStorage::doOpenRead");
+        }
+
+        return ChunkHandle.readHandle(chunkName);
+    }
+
+    @Override
+    protected ChunkHandle doOpenWrite(String chunkName) throws ChunkStorageException, IllegalArgumentException {
+        Path path = Paths.get(config.getRoot(), chunkName);
+        if (!Files.exists(path)) {
```

Comment:
please also check that this is a RegularFile, not a directory or a link (for instance)

Reply:
Good check to add. Adding

Reply:
Fixed